backends/tfhe-cuda-backend/cuda/src/integer/vector_find.cuh

```diff
@@ -0,0 +1,300 @@
+        });
+      }
+
+      generate_many_lut_device_accumulator<Torus>(
+          streams.stream(0), streams.gpu_index(0), mem_ptr->luts->get_lut(0, 0),
+          mem_ptr->luts->get_degree(0), mem_ptr->luts->get_max_degree(0),
+          mem_ptr->params.glwe_dimension, mem_ptr->params.polynomial_size,
+          mem_ptr->params.message_modulus, mem_ptr->params.carry_modulus, fns,
+          mem_ptr->allocate_gpu_memory);
+
+      mem_ptr->luts->broadcast_lut(streams.active_gpu_subset(1));
+
+      uint32_t block_modulus =
+          mem_ptr->params.message_modulus * mem_ptr->params.carry_modulus;
+      uint32_t box_size = mem_ptr->params.polynomial_size / block_modulus;
+      uint32_t current_lut_stride =
+          (block_modulus / current_chunk_size) * box_size;
+
+      integer_radix_apply_many_univariate_lookup_table<Torus>(
+          streams, mem_ptr->tmp_many_luts_output, current_selector, bsks,
+          (Torus *const *)ksks, mem_ptr->luts, current_chunk_size,
+          current_lut_stride);
+
+      copy_radix_ciphertext_slice_async<Torus>(
+          streams.stream(0), streams.gpu_index(0), current_output, chunk_idx,
+          chunk_idx + current_chunk_size, mem_ptr->tmp_many_luts_output, 0,
+          current_chunk_size);
+    }
+  }
+}
+
+template <typename Torus>
+uint64_t scratch_cuda_create_possible_results(
+    CudaStreams streams, int_possible_results_buffer<Torus> **mem_ptr,
+    int_radix_params params, uint32_t num_blocks, bool allocate_gpu_memory) {
+
+  uint64_t size_tracker = 0;
+  *mem_ptr = new int_possible_results_buffer<Torus>(
+      streams, params, num_blocks, allocate_gpu_memory, size_tracker);
+
+  return size_tracker;
+}
+
+template <typename Torus>
+__host__ void
+host_aggregate_one_hot_vector(CudaStreams streams,
+                              CudaRadixCiphertextFFI *lwe_array_out,
+                              CudaRadixCiphertextFFI const *lwe_array_in_list,
+                              uint32_t num_input_ciphertexts,
+                              int_aggregate_one_hot_buffer<Torus> *mem_ptr,
+                              void *const *bsks, Torus *const *ksks) {
+
+  int_radix_params params = mem_ptr->params;
+  uint32_t num_packed_blocks = mem_ptr->num_packed_blocks;
+  uint32_t chunk_size = mem_ptr->chunk_size;
+
+  CudaRadixCiphertextFFI *aggregated_vector = mem_ptr->aggregated_vector;
+  CudaRadixCiphertextFFI *temp_aggregated_vector =
+      mem_ptr->temp_aggregated_vector;
+  CudaRadixCiphertextFFI *message_ct = mem_ptr->message_ct;
+  CudaRadixCiphertextFFI *carry_ct = mem_ptr->carry_ct;
+
+  int_radix_lut<Torus> *identity_lut = mem_ptr->identity_lut;
+  int_radix_lut<Torus> *message_extract_lut = mem_ptr->message_extract_lut;
+  int_radix_lut<Torus> *carry_extract_lut = mem_ptr->carry_extract_lut;
+
+  uint32_t num_chunks = (num_input_ciphertexts + chunk_size - 1) / chunk_size;
+
+  for (uint32_t chunk_idx = 0; chunk_idx < num_chunks - 1; chunk_idx++) {
```

Comment:
These two loops can also be executed in parallel (see `tfhe/src/integer/server_key/radix_parallel/vector_find.rs:1285`). Would be good to use different streams to do this.